### ISSUE TITLE
Why restrict cross entropy to 2D inputs only?

### ISSUE THREAD

**Issue by hristo-vrigazov** [jonasrauber/eagerpy#57]
First, congrats on such a great project!

Basically the title. PyTorch and Tensorflow both support cross entropy in the ND case, and your implementation in Numpy would work for the multi-dimensional case too. However, in every function, there is an assert that the logits are 2D array. I propose to remove those asserts :smile:

**Comment by christian-westbrook** [comment]
I ran into this problem while using the latest version of https://github.com/bethgelab/foolbox, which uses eagerpy. I would also benefit from support for cross-entropy against more than two dimensions.
